src/NuGet.Clients/NuGet.PackageManagement.UI/GlobalSuppressions.cs

```diff
@@ -113,3 +113,4 @@
 [assembly: SuppressMessage("Build", "CA1501:'ButtonHyperlink' has an object hierarchy '9' levels deep within the defining module. If possible, eliminate base classes within the hierarchy to decrease its hierarchy level below '6': 'Hyperlink, Span, Inline, TextElement, FrameworkContentElement, ContentElement, DependencyObject, DispatcherObject, Object' (https://docs.microsoft.com/visualstudio/code-quality/ca1501-avoid-excessive-inheritance)", Justification = "<Pending>", Scope = "type", Target = "~T:NuGet.PackageManagement.UI.Controls.ButtonHyperlink")]
 [assembly: SuppressMessage("Build", "CA1501:'ButtonHyperlinkAutomationPeer' has an object hierarchy '8' levels deep within the defining module. If possible, eliminate base classes within the hierarchy to decrease its hierarchy level below '6': 'HyperlinkAutomationPeer, TextElementAutomationPeer, ContentTextAutomationPeer, FrameworkContentElementAutomationPeer, ContentElementAutomationPeer, AutomationPeer, DispatcherObject, Object' (https://docs.microsoft.com/visualstudio/code-quality/ca1501-avoid-excessive-inheritance)", Justification = "<Pending>", Scope = "type", Target = "~T:NuGet.PackageManagement.UI.Automation.ButtonHyperlinkAutomationPeer")]
 [assembly: SuppressMessage("Build", "CA1501:'ConfigurationFilesControl' has an object hierarchy '9' levels deep within the defining module. If possible, eliminate base classes within the hierarchy to decrease its hierarchy level below '6': 'UserControl, ContentControl, Control, FrameworkElement, UIElement, Visual, DependencyObject, DispatcherObject, Object'", Justification = "Default WPF class hierarchy", Scope = "type", Target = "~T:NuGet.PackageManagement.UI.Options.ConfigurationFilesControl")]
+[assembly: SuppressMessage("Usage", "VSTHRD003:Avoid awaiting foreign Tasks", Justification = "<Pending>", Scope = "member", Target = "~M:NuGet.PackageManagement.UI.PackageItemViewModel.GetDetailedPackageSearchMetadataAsync~System.Threading.Tasks.Task{System.ValueTuple{NuGet.VisualStudio.Internal.Contracts.PackageSearchMetadataContextInfo,NuGet.VisualStudio.Internal.Contracts.PackageDeprecationMetadataContextInfo}}")]
```

Comment:
I'm pretty sure Nikolche generally disagrees with me on this, but I would much prefer the suppressions be done inline with `#pragma` statements, rather than GlobalSuppressions.cs.

The reason is that when I'm working on some code with a supressed analyzer warning, I can immeidately fix the issue and reduce tech debt as part of my other work.  Similarly if someone else creates a PR and I see the `#pragma warning disable` just above or below their changes, I can comment asking them to make a small tech debt improvement, since they're already modifying that code anyway.

When suppressions are hidden in GlobalSuppression.cs, then it's much more difficult to know that there's even tech debt to improve.

Plus, global suppressions work on entire methods, so if someone adds new code that also violates the same analyzer in the same method, they won't get warned with GlobalSuppresions, but they will with inline suppressions.

Reply:
Ok, could you find a way to add this to our coding guidance so that we aren't informing PR authors that are already in code review?

Reply:
Moved all of the suppressions to pragma statements